### PR TITLE
Add country app with basic models

### PR DESCRIPTION
**Pull request by crypturgus**

### CODE REVIEW

File: api/company/models.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    user = models.OneToOneField(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        related_name="company",
+        help_text="The user that this company is associated with.",
+    )
+
+
+class Counterparty(Company):
+    pass
```
Comment (crypturgus) [inline api/company/models.py:43]: Missing relationship (m2m) to userprofile model?

File: api/company/models.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        help_text="The country where the company is located",
+    )
+
+    class Meta:
+        abstract = True
+
+
+class UserCompany(Company):
+    # TODO: Link it to UserProfile when there
+    user = models.OneToOneField(
```
Comment (crypturgus) [inline api/company/models.py:34]: what about UserProfile model?

File: api/company/models.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
+from django.conf import settings
+from django.db import models
+
+# TODO: django-countries?
+COUNTRY_CHOICES = [
+    ("PL", "Poland"),
+    ("US", "United States"),
+    ("UK", "United Kingdom"),
+    ("DE", "Germany"),
+    ("FR", "France"),
```
Comment (crypturgus) [inline api/company/models.py:10]: Czech Republic ;)

File: api/company/models.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    name = models.CharField(max_length=255)
+    address = models.TextField()
+    identification_number = models.CharField(
+        max_length=50,
+        unique=True,
+        help_text="Company's official identification number (e.g., NIP, VAT, EIN)",
+    )
+    country = models.CharField(
+        max_length=2,
+        choices=COUNTRY_CHOICES,
```
Comment (crypturgus) [inline api/company/models.py:24]: What do you think of this approach?:
https://docs.djangoproject.com/en/4.2/ref/models/fields/ -> `models.TextChoices`

Comment (lukaszsi) [reply api/company/models.py:24]: Yeah, that is pretty tight coupling of these values with models, while we could use these in other places potentially. Maybe even such mapping should have `country name / country code / currency` altogether taht would be then used in models, so we'd be sure we will have all necessary values for each new country?

Comment (lukaszsi) [reply api/company/models.py:16]: Check last commit, WDYT?

**Comment by crypturgus** [comment]
Looks good, but I can't validate, I don't know for what reason. Maybe it is because I created the PR or the PR has a `draft` label.

File: api/app/settings.py [hunk 5]
```diff
@@ -41,6 +41,7 @@
     "django.contrib.messages",
     "django.contrib.staticfiles",
     "user",
+    "company",
```
[the remark this answers is not in the document]
Comment (lukaszsi) [reply api/app/settings.py:44]: Sure, no hard opinion on that 👍

**Review by lukaszsi — APPROVED** [review]

**Pull request merged**